src/lib.ts

```diff
@@ -536,7 +545,7 @@ export class XarArchive {
         extractCertSection(this.signatureResources.cert),
         ...this.signatureResources.additionalCerts.map(extractCertSection)
       ];
-      let signatureSize = RSA_SIGNATURE_SIZE;
+      let signatureSize = signatureLength(this.signatureResources.privateKey);
```

Comment:
This might be premature optimization but signing is a non-trivial operation - can't you calculate signature size only once, e.g. when the private key is assigned?